packages/side-nav/src/vaadin-side-nav-i18n-mixin.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * @license
+ * Copyright (c) 2023 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+import { ChildrenController } from './vaadin-side-nav-children-controller.js';
+
+/**
+ * @polymerMixin
+ */
+export const SideNavI18nMixin = (superClass) =>
+  class SideNavI18nMixin extends superClass {
+    static get properties() {
+      return {
+        /**
+         * The object used to localize this component.
+         *
+         * To change the default localization, replace the entire
+         * `i18n` object with a custom one.
+         *
+         * The object has the following structure and default values:
+         * ```
+         * {
+         *   toggle: 'Toggle child items'
+         * }
+         * ```
+         *
+         * @type {SideNavI18n}
+         * @default {English/US}
+         */
+        i18n: {
+          type: Object,
+          value: () => {
+            return {
+              toggle: 'Toggle child items',
+            };
+          },
+        },
+
+        /**
+         * Count of child items.
+         * @private
+         */
+        _itemsCount: {
+          type: Number,
+          value: 0,
+        },
+      };
+    }
+
+    constructor() {
+      super();
+
+      this._childrenController = new ChildrenController(this, this._itemsSlotName);
+    }
+
+    /**
+     * Name of the slot to be used for children.
+     * @protected
+     */
+    get _itemsSlotName() {
+      return 'children';
+    }
+
+    /** @protected */
+    firstUpdated() {
+      super.firstUpdated();
+
+      // Controller that detects changes to the side-nav items.
+      this.addController(this._childrenController);
+    }
+
+    /**
+     * @protected
+     * @override
+     */
+    willUpdate(props) {
+      super.willUpdate(props);
+
+      const children = this._childrenController.nodes;
+      this._itemsCount = children.length;
+    }
+
+    /**
+     * @protected
+     * @override
+     */
+    updated(props) {
+      super.updated(props);
+
+      if (props.has('_itemsCount')) {
+        this.toggleAttribute('has-children', this._itemsCount > 0);
```

Comment:
This line seems a bit unrelated to the i18n mixin and would ideally be placed in a separate mixin. However, I think we can live with that for the time being.

Reply:
Renamed the mixin to `SideNavChildrenMixin` which seems to be a more suitable name.